### DISCUSSION TITLE
Required Properties Injection

### DISCUSSION THREAD

**Discussion by TonyValenti** [dadhi/DryIoc#562] [Ideas]
AutoFac recently implemented required properties injection and it is awesome!

https://autofac.readthedocs.io/en/latest/register/prop-method-injection.html#required-properties

This eliminates a ton of boilerplate constructor code, especially where inheritance is involved. 

Dryloc should follow suit.

**Comment by dadhi** [comment]
@TonyValenti Hi, yeah, I am aware of the new Autofac version with required properties.
I think we may safely change this discussion to the actual issue to work on.

**Reply by dadhi** [reply]
Done - #563
